src/lib.rs

```diff
@@ -2551,6 +2604,44 @@ impl Decoder {
                 }
                 Ok((Value::Seq(v), input))
             }
+            Decoder::RepeatFallback(subset, superset) => {
+                let mut accum = Vec::new();
+                let mut incr_input = input;
+
+                loop {
+                    match subset.parse(program, scope, incr_input) {
+                        Ok((b, next_input)) => {
+                            accum.push(b);
+                            incr_input = next_input;
+                        }
+                        Err(_err) => break,
+                    }
+                }
+
+                // track the last offset we reached to see if we get any further
+                let checkpoint = incr_input.offset;
+
+                loop {
+                    match superset.parse(program, scope, incr_input) {
+                        Ok((b, next_input)) => {
+                            accum.push(b);
+                            incr_input = next_input;
+                        }
+                        Err(_err) => break,
+                    }
+                }
+
+                if let Err(e) = (Decoder::EndOfInput).parse(program, scope, incr_input) {
```

Comment:
This is obviously wrong, but the MatchTree logic for how to judge effective end-of-RepeatFallback eludes me...